content/en/blog/_posts/2024-xx-xx-matchlabelkeys-podaffinity.md

```diff
@@ -0,0 +1,157 @@
+---
+layout: blog
+title: 'Kubernetes 1.31: MatchLabelKeys in PodAffinity graduates to beta'
+date: 2024-XX-XX
+slug: matchlabelkeys-podaffinity
+author: >
+  Kensei Nakada (Tetrate)
+---
+
+Kubernetes 1.29 introduced new fields `MatchLabelKeys` and `MismatchLabelKeys` in PodAffinity and PodAntiAffinity.
+
+In Kubernetes 1.31, this feature moves to beta and the corresponding feature gate (`MatchLabelKeysInPodAffinity`) gets enabled by default.
+
+## `MatchLabelKeys` - accurate scheduling during rolling upgrades
+
+During rolling upgrades, a cluster may have Pods from multiple versions of a Deployment. 
+This makes it challenging for the scheduler to distinguish between them based on the `LabelSelector` specified in PodAffinity or PodAntiAffinity. 
```

Comment:
```suggestion
However, the scheduler cannot distinguish between old and new versions based on the `LabelSelector` specified in PodAffinity or PodAntiAffinity. As a result, it will co-locate or disperse Pods regardless of their versions.
```